docs/02-providers/aws/01-setup.md

```diff
@@ -69,37 +69,132 @@ Default output format [None]: ENTER
 
 Credentials are stored in INI format in `~/.aws/credentials`, which you can edit directly if needed. Read more about that file in the [AWS documentation](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-config-files)
 
-You can even set up different profiles for different accounts, which can be used by Serverless as well. To specify a default profile to use, you can add a `profile` setting to your `provider` configuration in `serverless.yml`:
+You can even set up different profiles for different accounts, which can be used by Serverless as well.
 
+#### Specifying Credentials/Profiles to Serverless
+
+You can specify either credentials or a profile.  Each of these can be provided by altering your serverless.yml or your system's environment variables.  Each can be specified for all stages or you can specify stage specific credentials.  Using variables in your serverless.yml, you could implement more complex credential selection capabilities.
+
+
+One set of credentials for all stages using serverless.yml
 ```yml
-service: new-service
 provider:
-  name: aws
-  runtime: nodejs4.3
-  stage: dev
-  profile: devProfile
+  credentials:
+    accessKeyId: YOUR_ACCESS_KEY
+    secretAccessKey: YOUR_SECRET_KEY
 ```
 
-##### Per Stage Profiles
+A set of credentials for each stage using serverless.yml
+```yml
+vars:
+  test:
+    credentials:
+      accessKeyId: YOUR_ACCESS_KEY_FOR_TEST
+      secretAccessKey: YOUR_SECRET_KEY_FOR_TEST
+  prod:
+    credentials:
+      accessKeyId: YOUR_ACCESS_KEY_FOR_PROD
+      secretAccessKey: YOUR_SECRET_KEY_FOR_PROD
+provider:
+  credentials: ${self:vars.{opt:stage}.credentials}
+```
 
-As an advanced use-case, you can deploy different stages to different accounts by using different profiles per stage. In order to use different profiles per stage, you must leverage [variables](../../01-guide/08-serverless-variables.md) and the provider profile setting.
+One profile for all stages using serverless.yml
+```yml
+provider:
+  profile: your-profile
+```
+
+A profile for each stage using serverless.yml
+```yml
+vars:
```

Comment:
Currently vars doesn't work, it needs to be in ***custom***. Could you change all occurences of vars to custom for now (I think we should support vars in the future though